src/lib/command_line.ml

```diff
@@ -103,18 +103,36 @@ let run_pipeline
 
 let pipeline ~biokepi_machine ?work_directory =
   let parse_input_files files ~kind =
+    let open Biokepi.EDSL.Library.Input in
     let parse_file file prefix =
-      match Filename.check_suffix file ".bam" with
-      | true ->
-        Biokepi.EDSL.Library.Input.(
-          fastq_sample
-            ~sample_name:(prefix ^ "-" ^
-                          (Filename.chop_extension file |> Filename.basename))
-            [of_bam ~reference_build:"dontcare" `PE file]
-        )
-      | false ->
-        Yojson.Safe.from_file file
-        |> Biokepi.EDSL.Library.Input.of_yojson |> or_fail (prefix ^ "-json") in
+      let sample_name =
+        prefix ^ "-" ^ Filename.(chop_extension file |> basename)
+      in
+      let ends_with = Filename.check_suffix file in
+      (*
+        Beside serialized sample description files,
+        we also would like to capture direct paths to the BAMs/FASTQs
+        to make it easier for the user to submit samples.
+        Examples
+         - JSON file: /path/to/sample.json
+         - BAM file: https://url.to/my.bam
+         - Single-end FASTQ: /path/to/single.fastq.gz
+         - Paired-end FASTQ: /p/t/pair1.fastq;/p/t/pair2.fastq
+         ...
+      *)
+      if (ends_with ".bam") then begin
+        fastq_sample ~sample_name [of_bam ~reference_build:"dontcare" `PE file]
+      end
+      else if (ends_with ".fastq" || ends_with ".fastq.gz") then begin
+        match (String.split ~on:(`Character ';') file) with
```

Comment:
Maybe `:` or `@` would be better? `;` is a token in the shell's syntax which forces the user to quote everything.